### PR TITLE
TEC-4134: Delivery method selector

### PR DESCRIPTION
**Pull request by samuelvermeuln**
## Task link
[JIRA](https://evinobr.atlassian.net/jira/software/projects/TEC/boards/64?selectedIssue=TEC-4134&text=%5Bfront%5D)

## Task description
Delivery method selector

## How to test
 [Storybook](https://www.chromatic.com/build?appId=64e4f08113329cd99216030d&number=30)

## Ambient to test
`chromatic`

### CODE REVIEW

File: src/components/Card/Card.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <Card {...args} theme={theme} selected>
+    <Text color="#212425">Lorem Ipsum</Text>
+  </Card>
+)
+
+export const CardMaxWidth = (args: CardProps) => (
+  <Card {...args} theme={theme} selected maxWidth="50%">
+    <Text color="#212425">Lorem Ipsum</Text>
+  </Card>
+)
```
Comment (gabrielcdiniz) [inline src/components/Card/Card.stories.tsx]: convert to CSF pattern

- [Storybook Docs](https://storybook.js.org/docs/api/csf#default-export)

File: src/components/Card/style.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import styled from 'styled-components'
+import { Colors } from '../../tokens'
+import { CardProps } from './type'
+
+const StyledCard = styled.div<CardProps>`
+  display: flex;
+  max-width: ${(props: CardProps) => props.maxWidth || '100%'};
+  padding: 16px;
+  gap: 16px;
```
Comment (gabrielcdiniz) [inline src/components/Card/style.tsx:10]: get sizes by `Spacing` tokens

File: src/components/CardAddress/index.tsx [hunk 3]
```diff
@@ -12,45 +14,24 @@ interface CardAddressProps extends HTMLAttributes<HTMLInputElement> {
   complement?: string
 }
 
-const RadioButtonLabel = styled.label`
+const Flex = styled.div`
   display: flex;
   align-items: center;
+  gap: 16px;
```
Comment (gabrielcdiniz) [inline src/components/CardAddress/index.tsx:20]: get size by `Spacing` token

File: src/components/Line/Line.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+export const DefaultLine = (args: LineProps) => <Line {...args} />
+
+export const ColorLine = (args: LineProps) => <Line color="#307c27" {...args} />
+
+export const DashedLine = (args: LineProps) => <Line dashed {...args} />
+
+export const maxWidthLine = (args: LineProps) => (
+  <Line maxWidth="50%" {...args} />
+)
```
Comment (gabrielcdiniz) [inline src/components/Line/Line.stories.tsx]: convert to CSF pattern

- [Storybook Docs](https://storybook.js.org/docs/api/csf#default-export)

File: src/components/Line/style.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import styled from 'styled-components'
+import { LineProps } from './type'
+
+const StyledLine = styled.div<LineProps>`
+  width: 100%;
+  max-width: ${(props: LineProps) => props.maxWidth};
+  height: ${(props) => (props.dashed ? '1px' : '1px')};
```
Comment (gabrielcdiniz) [inline src/components/Line/style.tsx]: do we need this prop `dashed` ?

File: src/components/Radio/Radio.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  <Radio theme={theme} {...args} />
+)
+
+export const CheckedRadio = (args: RadioContentProps) => (
+  <Radio {...args} theme={theme} checked />
+)
+
+export const BigRadio = (args: RadioContentProps) => (
+  <Radio {...args} theme={theme} size="50px" />
+)
```
Comment (gabrielcdiniz) [inline src/components/Radio/Radio.stories.tsx]: convert to CSF pattern

- [Storybook Docs](https://storybook.js.org/docs/api/csf#default-export)

File: src/components/Radio/index.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  ...props
+}) => (
+  <RadioStyle
+    color={color}
+    theme={theme}
+    checked={checked}
+    {...props}
+    type="radio"
+  />
+)
```
Comment (gabrielcdiniz) [inline src/components/Radio/index.tsx]: maybe thus, do more "clean code"

```suggestion
const Radio: React.FC<RadioContentProps> = ({ ...props }) => (
  <RadioStyle {...props} type="radio"/>
)
```

File: src/components/Radio/style.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+          ? props.color || Colors[props.theme].action.action100
+          : Colors[props.theme].neutral.neutral100};
+    background-color: ${(props: RadioContentProps) =>
+      props.checked ? Colors[props.theme].neutral.neutral100 : 'transparent'};
+  }
+
+  &:hover {
+    &:checked {
+      border: 6px solid
+        ${(props) => props.color || Colors[props.theme].action.action200};
```
Comment (gabrielcdiniz) [inline src/components/Radio/style.tsx:31]: change only color

```suggestion
      border-color: ${(props) => props.color || Colors[props.theme].action.action200};
```

File: src/components/SelectListType/SelectListType.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+        <SelectListType.Title>TITLE</SelectListType.Title>
+        <SelectListType.SubTitle>Sub Title</SelectListType.SubTitle>
+        <SelectListType.Price>00,00 R$</SelectListType.Price>
+      </Flex>
+
+      <SelectListType.Action />
+    </SelectListType.Root>
+    <Line />
+  </>
+)
```
Comment (gabrielcdiniz) [inline src/components/SelectListType/SelectListType.stories.tsx]: convert to CSF pattern

- [Storybook Docs](https://storybook.js.org/docs/api/csf#default-export)

File: src/components/Text/Text.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+  return (
+    <>
+      {Color.map((value, index) => (
+        <Text color={Color[index]} {...args}>
+          Color <strong>{Color[index]}</strong>
+        </Text>
+      ))}
+    </>
+  )
+}
```
Comment (gabrielcdiniz) [inline src/components/Text/Text.stories.tsx]: convert to CSF pattern

- [Storybook Docs](https://storybook.js.org/docs/api/csf#default-export)

**Review by gabrielcdiniz — CHANGES_REQUESTED** [review]

File: src/components/Card/style.tsx (hunk 2 above)
Comment (samuelvermeuln) [reply src/components/Card/style.tsx:10]: This space cannot be the `Spacing` token, it is in rem and needs to be pix

File: src/components/CardAddress/index.tsx (hunk 3 above)
Comment (samuelvermeuln) [reply src/components/CardAddress/index.tsx:20]: This space cannot be the `Spacing` token, it is in rem and needs to be pix

File: src/components/Line/style.tsx (hunk 5 above)
Comment (samuelvermeuln) [reply src/components/Line/style.tsx]: not necessary just an option if necessary

**Review by gabrielcdiniz — APPROVED** [review]

**Comment by joelsalmeida** [comment]
Check the DOC and stories of Radio component. Maybe it's related to Gabriel's comment.

File: src/components/Radio/style.tsx (hunk 8 above)
Comment (samuelvermeuln) [reply src/components/Radio/style.tsx:31]: border: `border-width` `border-style` `border-color`

in this simplified way to properties

**Pull request merged**